MongoDBStitch.agent.lib.nut

```diff
@@ -0,0 +1,318 @@
+        _processRequest("POST", "/auth/session", null, { "isAuth" : true, "useRefreshToken" : true }, callback);
+    }
+
+    // Sends an http request to MongoDB Stitch service
+    function _processRequest(method, path, body, options, callback) {
+        local error = _validateNonEmptyArg(_appId, "appId");
+        if (error) {
+            _invokeCallback(error, null, callback);
+            return;
+        }
+
+        if (!options) {
+            options = {};
+        }
+        local url = _MONGO_DB_STITCH_BASE_URL + path;
+        local headers = {
+            "Accept" : "application/json",
+            "Content-Type" : "application/json"
+        };
+        local isAuth = _getTableValue(options, "isAuth", false);
+        local useRefreshToken = _getTableValue(options, "useRefreshToken", false);
+        if (!isAuth || useRefreshToken) {
+            local token = useRefreshToken ? _refreshToken : _accessToken;
+            if (!token) {
+                _invokeCallback(
+                    MongoDBStitchError(MONGO_DB_STITCH_ERROR.LIBRARY_ERROR, MONGO_DB_STITCH_UNAUTHORIZED),
+                    null,
+                    callback);
+                return;
+            }
+            headers.Authorization <- format("Bearer %s", token);
+        }
+        if (!isAuth && _getTableValue(options, "refreshOnFailure", null) == null) {
+            options.refreshOnFailure <- true;
+        }
+
+        _logDebug(format("Doing the request: %s %s, body: %s", method, url, http.jsonencode(body)));
+
+        local request = http.request(method, url, headers, body ? http.jsonencode(body) : "");
+        request.sendasync(function (response) {
+            _processResponse(response, method, path, body, options, callback);
+        }.bindenv(this));
+    }
+
+    // Processes http response from MongoDB Stitch service and executes callback if specified
+    function _processResponse(response, method, path, body, options, callback) {
+        _logDebug(format("Response status: %d, body: %s", response.statuscode, response.body));
+        local errType = null;
+        local errDetails = null;
+        local httpStatus = response.statuscode;
+        if (httpStatus < 200 || httpStatus >= 300) {
+            errType = MONGO_DB_STITCH_ERROR.MONGO_DB_STITCH_REQUEST_FAILED;
+            errDetails = format("%s: %i", MONGO_DB_STITCH_REQUEST_FAILED, httpStatus);
+        }
+        try {
+            response.body = (response.body == "") ? {} : http.jsondecode(response.body);
+        } catch (e) {
+            if (!errType) {
+                errType = MONGO_DB_STITCH_ERROR.MONGO_DB_STITCH_UNEXPECTED_RESPONSE;
+                errDetails = e;
+            }
+        }
+        
+        // try to refresh token in case it is expired
+        if (errType == MONGO_DB_STITCH_ERROR.MONGO_DB_STITCH_REQUEST_FAILED
+            && httpStatus == 401 && _getTableValue(response.body, "errorCode", null) == "InvalidSession"
```

Comment:
Use error_code here instead of errorCode

Reply:
@edaniels will take care of it, thanks!